libmariadb/mariadb_lib.c

```diff
@@ -1787,7 +1787,12 @@ MYSQL *mthd_my_real_connect(MYSQL *mysql, const char *host, const char *user,
  */
   if ((pkt_length=ma_net_safe_read(mysql)) == packet_error)
   {
-    if (mysql->net.last_errno == CR_SERVER_LOST)
+    if (mysql->options.use_ssl)
+      my_set_error(mysql, CR_CONNECTION_ERROR, SQLSTATE_UNKNOWN,
+                   "Received error packet before completion of TLS handshake. "
+                   "Suppressing its details so that the client cannot vary its behavior "
+                   "based on this UNTRUSTED input.");
+    else if (mysql->net.last_errno == CR_SERVER_LOST)
```

Comment:
May be still preserve the error message like the browser shows http pages without a lock sign?
For example,
```c
my_set_error(mysql, CR_CONNECTION_ERROR, SQLSTATE_UNKNOWN,
    "Received UNTRUSTED error packet before completion of TLS handshake. "
    "The authenticity of the following error cannot be verified: %d - %s",
    mysql->net.last_errno, mysql->net.last_error)
```

Reply:
That's a good idea. Will help to clearly identify and enumerate [any cases where the server sends a pre-TLS  error packet](https://jira.mariadb.org/browse/CONC-648?focusedCommentId=263341&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-263341).

Reply:
Done in https://github.com/mariadb-corporation/mariadb-connector-c/pull/223/commits/777e956b.